asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/JRubyAsciidoctor.java

```diff
@@ -507,8 +510,15 @@ private RubyClass getExtensionGroupClass() {
 
     @Override
     public void log(LogRecord logRecord) {
+        if (this.maxSeverity.compareTo(logRecord.getSeverity()) < 0) {
+            this.maxSeverity = logRecord.getSeverity();
```

Comment:
JRubyAsciidoctor can be used in a multithreaded way when embedded into other environments like Spring Boot apps or sth similar.
In such scenarios it would be dangerous to rely on this property.
Could this be moved into a custom `org.asciidoctor.log.LogHandler` that is only added by the AsciidoctorInvoker?

Reply:
Ok, moved this to MaxSeverityLogHandler under org.asciidoctor.jruby.cli package.